src/lib/editable-feature-collection.js

```diff
@@ -200,54 +209,124 @@ export class EditableFeatureCollection {
-        handles = handles.concat(
-          getEditHandles(geometry.coordinates, [], includeIntermediate, featureIndex)
+          break;
+        case 'MultiPolygon':
+          // positions are nested 3 levels
+          for (let a = 0; a < geometry.coordinates.length; a++) {
+            for (let b = 0; b < geometry.coordinates[a].length; b++) {
+              lineStringTupleArray.push({
+                featureIndex,
+                coordinates: geometry.coordinates[a][b],
+                positionIndexPrefix: [a, b]
+              });
+            }
+          }
+          break;
+        default:
+          throw Error(`Unhandled geometry type: ${geometry.type}`);
+      }
+    });
+
+    let intermediatePoint = null;
+    // the index of tuple array of the line / feature closest to the reference point
+    let tupleIndex = null;
+
+    const referencePoint = groundCoords && toPoint(groundCoords);
+
+    lineStringTupleArray.forEach(({ featureIndex, coordinates, positionIndexPrefix }, i) => {
+      if (referencePoint) {
+        // calculate the intermediate point location by determining the closest linestring segment
+        const lineStringFeature = toLineString(coordinates);
+        const candidateIntermediatePoint = nearestPointOnLine(lineStringFeature, referencePoint);
+        const nearestControlPoint = nearestPoint(
+          referencePoint,
+          toFeatureCollection(coordinates.map(c => toPoint(c)))
         );
-        break;
-      case 'Polygon':
-      case 'MultiLineString':
-        // positions are nested 2 levels
-        for (let a = 0; a < geometry.coordinates.length; a++) {
-          handles = handles.concat(
-            getEditHandles(geometry.coordinates[a], [a], true, featureIndex)
-          );
+
+        if (
+          // find the best candidate point
+          (!intermediatePoint ||
+            candidateIntermediatePoint.properties.dist < intermediatePoint.properties.dist) &&
+          // prohibit intermediate point from displaying when it is too close to an edit handle
+          project &&
+          getPixelDistanceBetweenPoints(
+            project(candidateIntermediatePoint.geometry.coordinates),
+            project(nearestControlPoint.geometry.coordinates)
+          ) > 10
+        ) {
+          intermediatePoint = candidateIntermediatePoint;
+          tupleIndex = i;
         }
-        break;
-      case 'MultiPolygon':
-        // positions are nested 3 levels
-        for (let a = 0; a < geometry.coordinates.length; a++) {
-          for (let b = 0; b < geometry.coordinates[a].length; b++) {
-            handles = handles.concat(
-              getEditHandles(geometry.coordinates[a][b], [a, b], true, featureIndex)
-            );
-          }
+      }
+
+      // convert the linestring / coordinates to edit handles
+      handles = [...handles, ...getEditHandles(coordinates, positionIndexPrefix, featureIndex)];
+    });
+
+    // add the lone intermediate point
+    if (
```

Comment:
Looks like if your point is over an existing edit handle but also close enough to a line, the intermediate point is shown. We should only render an intermediate point if, and only if, dragging from that point will create a new intermediate point.